tools/testsocket/testsocket.cpp

```diff
@@ -343,12 +343,19 @@ int readResults(ISocket * socket, bool readBlocked, bool useHTTP, StringBuffer &
             {
                 response = (const char *)mb.readDirect(len);
                 responseTree.setown(createPTreeFromXMLString(len, response));
+                assertex(responseTree);
             }
             else if (strieq("json", outputFmtStr))
             {
                 response = (const char *)mb.readDirect(len);
-                responseTree.setown(createPTreeFromJSONString(len, response));
+                // JCSMORE - json string coming back from IXmlWriterExt is always rootless the moment, so workaround it by supplying ptr_noRoot to reader
+                // writer should be fixed.
```

Comment:
Opened JIRA related to this issue : https://track.hpccsystems.com/browse/HPCC-18831